src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -138,3 +139,36 @@ export const useIsExecutionLoop = (): boolean => {
   const { safeAddress } = useSafeInfo()
   return wallet ? sameString(wallet.address, safeAddress) : false
 }
+
+export const useRecommendedNonce = (
+  safeTx?: SafeTransaction,
+):
+  | {
+      nonce: number
+      safeTxGas: number
+    }
+  | undefined => {
+  const { safe } = useSafeInfo()
+  const safeTxData = safeTx?.data
+
+  // Memoize only the necessary params, so that it doesn't change every time safeTx is changed
+  const safeTxParams = useMemo(
+    () =>
+      safeTxData?.to
+        ? {
+            to: safeTxData.to,
+            value: safeTxData.value,
+            data: safeTxData.data,
+            operation: safeTxData.operation,
+          }
+        : undefined,
+    [safeTxData?.to, safeTxData?.value, safeTxData?.data, safeTxData?.operation],
+  )
+
+  const [recommendedParams] = useAsync(() => {
+    if (!safeTxParams) return
+    return getRecommendedTxParams(safeTxParams)
```

Comment:
Have you read the comment for `safeTxParams`? 

```
  // Memoize only the necessary params, so that it doesn't change every time safeTx is changed
```

Reply:
Ah, that makes sense, thanks!